src/challenges.js

```diff
@@ -1,41 +1,80 @@
 // Desafio 1
-function compareTrue() {
-  // seu código aqui
+function compareTrue(boolean1, boolean2) {
+  if (boolean1 && boolean2 === true) {
+    return true;
+  } return false;
 }
 
 // Desafio 2
-function calcArea() {
-  // seu código aqui
+function calcArea(base, height) {
+  return (base * height) / 2;
 }
 
 // Desafio 3
-function splitSentence() {
-  // seu código aqui
+function splitSentence(string) {
+  return string.split(' ');
 }
 
 // Desafio 4
-function concatName() {
-  // seu código aqui
+function concatName(array) {
+  let newString = '';
+  newString += array[array.length - 1];
+  newString += ', ';
+  newString += array[0];
+  return newString;
 }
 
 // Desafio 5
-function footballPoints() {
-  // seu código aqui
+function footballPoints(wins, ties) {
+  let winPoints = wins * 3;
+  let tiePoints = ties * 1;
+  return winPoints + tiePoints;
 }
 
 // Desafio 6
-function highestCount() {
-  // seu código aqui
+function highestCount(array) {
+  let biggestNumber = 0;
+  let highestNumberCount = 0;
+
+  for (let number in array) {
+    if (array[number] > biggestNumber) {
+      biggestNumber = array[number];
+      highestNumberCount = 1;
+    } else if (array[number] === biggestNumber) {
+      highestNumberCount += 1;
+    }
+  }
+  return highestNumberCount;
 }
 
 // Desafio 7
-function catAndMouse() {
-  // seu código aqui
+function catAndMouse(mouse, cat1, cat2) {
+  let distanceCat1 = Math.abs(cat1 - mouse);
+  let distanceCat2 = Math.abs(cat2 - mouse);
+
+  if (distanceCat1 === distanceCat2) {
+    return 'os gatos trombam e o rato foge';
+  } else if (distanceCat1 < distanceCat2) {
+    return 'cat1';
+  }
+  return 'cat2';
 }
 
 // Desafio 8
-function fizzBuzz() {
-  // seu código aqui
+function fizzBuzz(array) {
```

Comment:
Oi @williamjog respondendo sua dúvida lá no Slack 

```suggestion
function parseNumberToFizzBuzz(number) {
    if (array[number] % 3 === 0 && array[number] % 5 === 0) {
        return 'fizzBuzz';
    } else if (array[number] % 3 === 0) {
        return 'fizz';
    } else if (array[number] % 5 === 0) {
        return 'buzz';
    } else {
        return 'bug!';
    }
}

function fizzBuzz(array) {
    for (let number in array) {
        array[num1] = parseNumberToFizzBuzz(number);
    }
    return newArray;
  }	
}
```

Reply:
@williamjog o código acima tem algumas falhas que você vai encotrar, a ideia que quis passar é mais da refatoração mas aí fica um desafio para você fazer as devidas correções.

Reply:
@renatosousafilho Muito obrigado, Renato! Vou tentar refatorar, sim! Eu tinha tentando fazer algo semelhante com função mas não encapsulei o suficiente.